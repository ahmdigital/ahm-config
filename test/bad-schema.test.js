const config = require('../lib');

describe('ahm-config: bad-schema', () => {
  const path = `${__dirname}/bad-schema`;

  it('should return schema errors when normalise disabled', () =>
    new Promise((done) => {
      const onError = (err) => {
        expect(err.message).toEqual(
          "Config error: undefined is not of a type(s) integer. Value 'undefined' should be undefined.",
        );
        expect(err.meta).toMatchSnapshot();
        done();
      };
      config.make({ path, normalise: false, onError });
    }));

  it('should return schema errors when normalise enabled', () =>
    new Promise((done) => {
      const onError = (err) => {
        expect(err.meta).toMatchSnapshot();
        done();
      };
      config.make({ path, normalise: true, onError });
    }));
});
