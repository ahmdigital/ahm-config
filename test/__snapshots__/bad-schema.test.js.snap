// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ahm-config: bad-schema should return schema errors when normalise disabled 1`] = `
Array [
  ValidationError {
    "argument": Array [
      "integer",
    ],
    "instance": "asd",
    "message": "is not of a type(s) integer",
    "name": "type",
    "path": Array [
      "multyKeyProp",
      "data",
      "prop1",
    ],
    "property": "instance.multyKeyProp.data.prop1",
    "schema": Object {
      "type": "integer",
    },
    "stack": "instance.multyKeyProp.data.prop1 is not of a type(s) integer",
  },
  Object {
    "instance": "/a",
    "message": "Has special characters in it (/^[a-z0-9_\\\\-$]+$/i are allowed)",
    "name": "key",
    "property": "instance./a",
  },
  Object {
    "instance": "%x",
    "message": "Has special characters in it (/^[a-z0-9_\\\\-$]+$/i are allowed)",
    "name": "key",
    "property": "instance.%x",
  },
]
`;

exports[`ahm-config: bad-schema should return schema errors when normalise enabled 1`] = `
Array [
  ValidationError {
    "argument": Array [
      "integer",
    ],
    "instance": NaN,
    "message": "is not of a type(s) integer",
    "name": "type",
    "path": Array [
      "multyKeyProp",
      "data",
      "prop1",
    ],
    "property": "instance.multyKeyProp.data.prop1",
    "schema": Object {
      "type": "integer",
    },
    "stack": "instance.multyKeyProp.data.prop1 is not of a type(s) integer",
  },
  Object {
    "instance": "/a",
    "message": "Has special characters in it (/^[a-z0-9_\\\\-$]+$/i are allowed)",
    "name": "key",
    "property": "instance./a",
  },
  Object {
    "instance": "%x",
    "message": "Has special characters in it (/^[a-z0-9_\\\\-$]+$/i are allowed)",
    "name": "key",
    "property": "instance.%x",
  },
]
`;
